### PR TITLE
Add a blog post describing the 1.4/1.5 release timeline and features

### PR DESCRIPTION
**Pull request by aljoscha**

### CODE REVIEW

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+## Lower Latency via Improvements to the Apache Flink Network Stack
+
+Speed will always be a key consideration for users who build stream processing applications, and
+Flink 1.5 will include a rework of the network stack that will even further improve Flink's latency.
+At the heart of this work is a transition to event-driven network I/O, ensuring that Flink will use
+all available network capacity, as well as credit-based flow control which offers more fine-grained
+backpressuring for improved checkpoint alignments.
+
+In our testing ([see slide 26 here](https://www.slideshare.net/FlinkForward/flink-forward-berlin-2017-nico-kruber-building-a-network-stack-for-optimal-throughput-lowlatency-tradeoffs)),
+we’ve seen a substantial improvement in latency using event- driven network I/O, and the community
```
Comment (zentol) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: remove spacer after "event-"

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+Flink 1.5 will introduce task-local recovery, which means that Flink will store a second copy of the
+most recent checkpoint on the local disk (or even in main memory) of a task manager. The primary
+copy still goes to durable storage so that it’s resilient to machine failures.
+
+In case of failover, the scheduler will try to reschedule tasks to their previous task manager (in
+other words, to the same machine again) if this is possible. The task can then recover from the
+locally-kept state. This makes it possible to avoid reading all state from the distributed file
+system (which is remote over the network). Especially in applications with very large state, not
+having to read many gigabytes over the network and instead from local disk would result in
```
Comment (zentol) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: would -> will?

File: content/news/2017/08/05/release-1.3.2.html [hunk 3]
```diff
@@ -139,7 +139,8 @@
       <h1>Apache Flink 1.3.2 Released</h1>
 
       <article>
-        <p>05 Aug 2017</p>
+        <p>05 Aug 2017
```
Comment (zentol) [inline content/news/2017/08/05/release-1.3.2.html]: would be neat if you could revert these; the next regeneration may reverse them again.

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+model and distributed process. The underlying motivation for FLIP-6 was the fact that Flink is being
+adopted by a wider range of developer communities--both developers coming from the big data and
+analytics space as well as developers coming from the event-driven applications space.
+
+Modern, stateful stream processing has served as a convergence for these two developer communities.
+Despite a significant overlap of the core concepts in the applications being built, each group of
+developers has its own set of common tools, deployment models, and expected behavior when working
+with a stream processing framework like Flink.
+
+FLIP-6 will ensure that Flink fits naturally in both of these contexts, behaving although it’s
```
Comment (HolyHaddock) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: "as though"?

**Comment by aljoscha** [comment]
Thanks for the comments, @zentol and @HolyHaddock, I addressed them.

@zentol That change with the `</p>` tags was caused by the multiple-authors change, I fixed that as well.

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  name: "Aljoscha Krettek"
+  twitter: "aljoscha"
+- mike:
+  name: "Mike Winters"
+  twitter: "wints"
+---
+
+The Apache Flink 1.4.0 release is on track to happen in the next couple of weeks, and for all of the
+readers out there who haven’t been following the release discussion on [Flink’s developer mailing
+list](http://flink.apache.org/community.html#mailing-lists), we’d like to provide some detail on
```
Comment (bowenli86) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: detail**s**

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+[FLIP-6](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=65147077) is an initiative
+that’s been in the works for more than a year and represents a major refactor of Flink’s deployment
+model and distributed process. The underlying motivation for FLIP-6 was the fact that Flink is being
+adopted by a wider range of developer communities--both developers coming from the big data and
+analytics space as well as developers coming from the event-driven applications space.
+
+Modern, stateful stream processing has served as a convergence for these two developer communities.
+Despite a significant overlap of the core concepts in the applications being built, each group of
+developers has its own set of common tools, deployment models, and expected behavior when working
```
Comment (bowenli86) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: behavior**s**

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+with a stream processing framework like Flink.
+
+FLIP-6 will ensure that Flink fits naturally in both of these contexts, behaving as though it’s
```
Comment (bowenli86) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: in both contexts and behaves as though....

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+[rescalable state](https://data-artisans.com/blog/apache-flink-at-mediamath-rescaling-stateful-applications),
+lays the groundwork for dynamic scaling in Flink, meaning that Flink programs will be able to scale up or down
+automatically based on required resources--a huge step forward in terms of ease of operability and
+the efficiency of Flink applications.
+
+## Lower Latency via Improvements to the Apache Flink Network Stack
+
+Speed will always be a key consideration for users who build stream processing applications, and
+Flink 1.5 will include a rework of the network stack that will even further improve Flink's latency.
+At the heart of this work is a transition to event-driven network I/O, ensuring that Flink will use
```
Comment (bowenli86) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: can you explain here this is a transition from what to event-driven network I/O?

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+This post will describe how the community plans to get there and the rationale behind the approach.
+
+## Coming soon: Major Changes to the Flink’s Runtime
+
+There are 3 significant improvements to the Apache Flink engine that the community has nearly
+completed and that will have a meaningful impact on Flink’s operability and performance.
+
+1. Rework of the deployment model and distributed process
+2. Transition to event-driven network I/O and application-level flow control for better backpressure handling
```
Comment (bowenli86) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: can you explain here this is a transition from what to event-driven network I/O?

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 5 above)
Comment (aljoscha) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: fixing

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 9 above)
Comment (wints) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: @aljoscha @bowenli86 

A proposed change:

"Transition from configurable, fixed-interval network I/O to event-driven network I/O and application-level flow control for better backpressure handling"

In the status quo, Flink uses a static output flusher that flushes buffers that aren't full every 100ms (this time interval is configurable). In the future, there'd be no more configuration parameter and Flink would get the best possible performance based on resources available at any given time.

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 8 above)
Comment (wints) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: @aljoscha @bowenli86 proposed change to this section, similar to my comment above. 

"At the heart of this work is a transition from configurable, fixed-interval network I/O to event-driven network I/O and application-level flow control, ensuring that Flink..."

**Comment by aljoscha** [comment]
I pushed changes to address those comments.

**Comment by bowenli86** [comment]
👍

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+what’s coming in Flink 1.4.0 as well as a preview of what the Flink community will save for 1.5.0.
+
+Both releases include ambitious features that we believe will move Flink to an entirely new level in
+terms of the types of problems it can solve and applications it can support. The community deserves
+lots of credit for its hard work over the past few months, and we’re excited to see these features
+in the hands of users.
+
+This post will describe how the community plans to get there and the rationale behind the approach.
+
+## Coming soon: Major Changes to the Flink’s Runtime
```
Comment (rmetzger) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: "Major changes to Flink's Runtime" (remove the "the")

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+There are 3 significant improvements to the Apache Flink engine that the community has nearly
+completed and that will have a meaningful impact on Flink’s operability and performance.
+
+1. Rework of the deployment model and distributed process
+2. Transition from configurable, fixed-interval network I/O to event-driven network I/O and application-level flow control for better backpressure handling
+3. Faster recovery from failure
+
+Next, we’ll go through each of these improvements in more detail.
+
+## Reworking Flink’s Deployment Model and Distributed Process
```
Comment (rmetzger) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: "Distributed Processing"?

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+[FLIP-6](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=65147077) is an initiative
```
Comment (rmetzger) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: I would explain what "FLIP" stands for.

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+## Lower Latency via Improvements to the Apache Flink Network Stack
+
+Speed will always be a key consideration for users who build stream processing applications, and
+Flink 1.5 will include a rework of the network stack that will even further improve Flink's latency.
+At the heart of this work is a transition from configurable, fixed-interval network I/O to event-
+driven network I/O and application-level flow control, ensuring that Flink will use all available
+network capacity, as well as credit-based flow control which offers more fine-grained backpressuring
+for improved checkpoint alignments.
+
+In our testing ([see slide 26 here](https://www.slideshare.net/FlinkForward/flink-forward-berlin-2017-nico-kruber-building-a-network-stack-for-optimal-throughput-lowlatency-tradeoffs)),
```
Comment (rmetzger) [inline _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: Afaik Slideshare supports linking to individual slides immediately, like this:
https://www.slideshare.net/FlinkForward/flink-forward-berlin-2017-nico-kruber-building-a-network-stack-for-optimal-throughput-lowlatency-tradeoffs#26

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 10 above)
Comment (aljoscha) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: fixing

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 11 above)
Comment (aljoscha) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: @wints what do you think?

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 12 above)
Comment (aljoscha) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: I think  going to the link explains it. Or we could just call the link "design document".

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 13 above)
Comment (aljoscha) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: fixing

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 12 above)
Comment (aljoscha) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: adding a comment in parentheses

File: _posts/2017-11-21-release-1.4-and-1.5-timeline.md (hunk 11 above)
Comment (wints) [reply _posts/2017-11-21-release-1.4-and-1.5-timeline.md]: +1 to changing to "distributed processing"

**Pull request closed without merging**